crates/source_analyzer/src/cycler.rs

```diff
@@ -0,0 +1,255 @@
+use std::{
+    collections::{HashMap, HashSet},
+    fmt::{Display, Formatter},
+    path::Path,
+};
+
+use itertools::Itertools;
+use serde::Deserialize;
+use topological_sort::TopologicalSort;
+
+use crate::{
+    contexts::Field,
+    error::Error,
+    manifest::{CyclerManifest, FrameworkManifest},
+    node::Node,
+};
+
+pub type CyclerName = String;
+pub type InstanceName = String;
+pub type ModulePath = String;
+
+#[derive(Debug)]
+pub struct Instance {
+    pub name: InstanceName,
+}
+
+impl Display for Instance {
```

Comment:
Understood, maybe remove the `Display` stuff because it is a lot of visual noise and only used rarely and we might ditch the Pepsi dump commands in the future?

Reply:
I grasp two different aspects in this discussion:

1. Having a `type Instance = String` instead of a `struct String`
This might be an artifact of previous iterations (I agree). While writing the code I understood the concept of an instance as more than just its name. In the end it boiled down to having just one property (`name`). Being strict this might additionally be YAGNI as an `Instance` might have additional properties. The reason to already use two distinct types for the instance itself and the name is that in the source analysis and code generation these concepts are treated differently.

2. Removing Display
I agree that this is only for debug purposes using `pepsi dump[...]` I am going to move the implementation to `pepsi` but vote to not delete it as the information about sorted nodes is otherwise not available to the user

Reply:
1. Vote for `type Instance = String`
2. Ok, yes, moving it near the usage is also good.

Looks good now, resolving.